### ISSUE TITLE
Python 3.6.7: ImportError: cannot import name 'filter_traceback'

### ISSUE THREAD

**Issue by dstromberg** [htm-community/nupic.cpp#489]
I built and tried to run nupic.cpp using Python 2.7, and then 3.6.  2.7 went very well.

For 3.6, it appeared to build, and the c++ tests only had one failure.

Sadly, the Python unit tests failed to check much, ending with:
```
copying build/lib.linux-x86_64-3.6/nupic/dummy.cpython-36m-x86_64-linux-gnu.so -> src/nupic
Traceback (most recent call last):
  File "setup.py", line 57, in <module>
    
  File "/usr/lib/python3.6/runpy.py", line 263, in run_path
    pkg_name=pkg_name, script_name=fname)
  File "/usr/lib/python3.6/runpy.py", line 96, in _run_module_code
    mod_name, mod_spec, pkg_name, script_name)
  File "/usr/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/dstromberg/src/nupic/nupic.cpp/nupic.cpp/bindings/py/packaging/setup.py", line 286, in <module>
    "nupic-bindings-check = nupic.bindings.check:checkMain",
  File "/home/dstromberg/.local/lib/python3.6/site-packages/setuptools/__init__.py", line 145, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/dstromberg/.local/lib/python3.6/site-packages/setuptools/command/test.py", line 229, in run
    self.run_tests()
  File "/home/dstromberg/src/nupic/nupic.cpp/nupic.cpp/bindings/py/packaging/setup.py", line 123, in run_tests
    import pytest
  File "/home/dstromberg/.local/lib/python3.6/site-packages/pytest.py", line 21, in <module>
    from _pytest.config import (
  File "/home/dstromberg/.local/lib/python3.6/site-packages/_pytest/config/__init__.py", line 30, in <module>
    from _pytest._code import filter_traceback
ImportError: cannot import name 'filter_traceback'

```
Is this a known issue?  Is there a fix?

Thanks!

**Comment by dstromberg** [comment]
I suspect a too-new version of pytest.

**Comment by dstromberg** [comment]
I pegged pytest with pytest==3.0.7 and got the tests to run.  They're failing, but that's another matter.

**Comment by breznak** [comment]
Thanks for reporting! 

>  pegged pytest with pytest==3.0.7 and got the tests to run. 

interesting, I have locally also 3.0.7, ideally we fix this, but might just force the exact version

> They're failing, but that's another matter.

that is strange, is your install correct? You can see py3 CI (i think it's OSX) with tests passing (welp, it was failing recently due to some unrelated network error in the CI)

**Comment by dan-stromberg** [comment]
I suspect the problem is 2.x binaries remaining after I tried to switch to 3.x.

I'm getting several: algorithms.so: undefined symbol: _Py_ZeroStruct

I tried "python2 setup.py clean" followed by "python3 setup.py clean", but that didn't work.
So I tried removing all the .o's and .so's in my nupic.cpp hierarchy using find | xargs, but that didn't work either.

What's the approved way of erasing intermediate files and starting over with a new version of Python?
